WooCommerce/Classes/ServiceLocator/ProductImageUploader.swift

```diff
@@ -278,16 +352,43 @@ private extension ProductImageUploader {
         }
     }
 
+    private func observeStatuses() {
+        userDefaultsStatuses.statusesPublisher
+            .sink { [weak self] statuses in
+                guard let self = self else { return }
+
+                // Handle all the errors
+                let failureStatuses = statuses.filter({ $0.isUploadFailure })
+                for failureStatus in failureStatuses {
+                    if let error = failureStatus.error, let asset = failureStatus.asset {
+                        let errorInfo = ProductImageUploadErrorInfo(
+                            siteID: failureStatus.siteID,
+                            productOrVariationID: failureStatus.productOrVariationID,
+                            error: .failedUploadingImage(asset: asset, error: error)
+                        )
+                        self.errorsSubject.send(errorInfo)
+                    }
+                }
+            }
+            .store(in: &cancellables)
+    }
+
     func observeStatusUpdates(key: Key, actionHandler: ProductImageActionHandler) {
         let observationToken = actionHandler.addUpdateObserver(self) { [weak self] productImageStatuses in
             guard let self = self else { return }
 
-            if !activeUploadsPublisher.contains(key), productImageStatuses.hasPendingUpload {
-                activeUploadsPublisher.append(key)
-            } else if activeUploadsPublisher.contains(key), !productImageStatuses.hasPendingUpload {
-                /// When all pending uploads are completed or removed,
-                /// remove the key from active uploads
-                removeProductFromActiveUploads(key: key)
+            if featureFlagService.isFeatureFlagEnabled(.backgroundProductImageUpload) {
+                // Update the states in userDefaultsStatuses
+                self.userDefaultsStatuses.appendStatuses(productImageStatuses, for: key.siteID, productID: key.productOrVariationID)
+            }
+            else {
+                if !activeUploadsPublisher.contains(key), productImageStatuses.hasPendingUpload {
+                    activeUploadsPublisher.append(key)
+                } else if activeUploadsPublisher.contains(key), !productImageStatuses.hasPendingUpload {
+                    /// When all pending uploads are completed or removed,
+                    /// remove the key from active uploads
+                    removeProductFromActiveUploads(key: key)
+                }
```

Comment:
Since you're maintaining `userDefaultsStatuses` disregarding the feature flag following my suggestion, the feature flag check should be removed here. 

Also, are you sure you want to append every new status in the storage? Consider this case:
- An image is uploaded, you receive the first status being `uploading` for a given product.
- When the upload completes, you get a new status for the same image, this time it's `remote`.

Appending every new status means the storage keeps the `uploading` status even after the upload completes.

So instead, you should replace all the statuses in the storage for a given product with the new ones you received in this observation.

Reply:
Thanks for the explanations, sound good to me!